src/python/Publisher/utils.py

```diff
@@ -0,0 +1,92 @@
+import os
+import time
+import logging
+import hashlib
+import subprocess
+
+from WMCore.Services.SiteDB.SiteDB import SiteDBJSON
+from WMCore.Credential.Proxy import Proxy
+
+__version__ = '1.0.3'
+
+def getHashLfn(lfn):
+    """
+    Provide a hashed lfn from an lfn.
+    """
+    return hashlib.sha224(lfn).hexdigest()
+
+def getFTServer(site, view, db, log):
+    """
+    Parse site string to know the fts server to use
+    """
+    country = site.split('_')[1]
+    query = {'key':country}
+    try:
+        fts_server = db.loadView('asynctransfer_config', view, query)['rows'][0]['value']
+    except IndexError:
+        log.info("FTS server for %s is down" % country)
+        fts_server = ''
+    return fts_server
+
+def execute_command(command):
+    """
+    _execute_command_
+    Function to manage commands.
+    """
+    proc = subprocess.Popen(
+           ["/bin/bash"], shell=True, cwd=os.environ['PWD'],
+           stdout=subprocess.PIPE,
+           stderr=subprocess.PIPE,
+           stdin=subprocess.PIPE,
+    )
+    proc.stdin.write(command)
+    stdout, stderr = proc.communicate()
+    rc = proc.returncode
+
+    return stdout, stderr, rc
+
+def getDNFromUserName(username, log, ckey = None, cert = None):
```

Comment:
Doesn't this function exist already ? Otherwise maybe better to keep it anyhow in the same place as getUserNameFromDN which surely CRAB must be using

Reply:
Not on top of my mind at with a quick look. But this is good imho, it is basically calling the wmcore function